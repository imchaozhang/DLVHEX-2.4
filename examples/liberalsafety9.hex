% If evaluated with --heuristics=monolithic and --domainexpansionsafety,
% this program tests if it works that outer external atoms become inner ones.
% While forbidden with strong safety, this should work with domainexpansionsafety.

% In contrast to liberalsafety6.hex, here the domain exploration algorithm
% never needs to guess the assignments to q(a), because the parameter is stratified.
q(a).
p(X) :- &id[q](X).
res(Y) v res2(Y) :- p(X), &testConcat[X,a](Y).
