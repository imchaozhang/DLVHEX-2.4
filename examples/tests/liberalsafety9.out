{q(a),p(a),res(aa)}
{q(a),p(a),res2(aa)}
